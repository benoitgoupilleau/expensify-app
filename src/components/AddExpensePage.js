import React from 'react';
import { connect } from 'react-redux';

import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';

// class AddExpensePage extends React.Component {
//   onSubmit = (expense) => {
//     this.props.addExpense(expense);
//     this.props.history.push('/');
//   };

//   render() {
//     return (
//       <div>
//         <h1>Add Expense</h1>
//         <ExpenseForm
//           buttonMessage="Add Expense"
//           onSubmit={this.onSubmit}
//         />
//       </div>
//     );
//   }
// }


export const AddExpensePage = (props) => ( // export for test
  <div>
    <h1>Add Expense</h1>
    <ExpenseForm
      buttonMessage="Add Expense"
      onSubmit={(expense) => {
        props.startAddExpense(expense);
        props.history.push('/dashboard');
      }}
    />
  </div>
);

export default connect(null, { startAddExpense })(AddExpensePage);
