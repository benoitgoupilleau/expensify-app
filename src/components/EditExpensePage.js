import React from 'react';
import { connect } from 'react-redux';

import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';

export const EditExpensePage = (props) => { // export for test
  return (
    <div className="page-container">
      <div className="page-header">
        <div className="content-container" >
          <h1 className="page-header__title">Edit Expense</h1>
        </div>
      </div>
      <div className="content-container">
        <ExpenseForm
          expense={props.expense}
          buttonMessage="Save Expense"
          onSubmit={(expense) => {
            props.startEditExpense(props.match.params.id, expense);
            props.history.push('/dashboard');
          }}
        />
        <button
          className="button button--secondary"
          onClick={() => {
            props.startRemoveExpense({ id: props.match.params.id });
            props.history.push('/dashboard');
          }}
        >Remove Expense
        </button>
      </div>
    </div>
  );
};

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  };
};

export default connect(mapStateToProps, { startEditExpense, startRemoveExpense })(EditExpensePage);
