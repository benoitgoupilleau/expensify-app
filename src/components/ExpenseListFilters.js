import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters';

export class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null
  };
  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };
  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  };
  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text} 
          onChange={(e) => {
            this.props.setTextFilter(e.target.value);
          }}
        />
        <select
          value={this.props.filters.sortBy} 
          onChange={(e) => {
            if (e.target.value === 'date') this.props.sortByDate();
            if (e.target.value === 'amount') this.props.sortByAmount();
          }}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          firstDayOfWeek={1}
          showClearDates={true}
          isOutsideRange={(day) => false}
          displayFormat="DD/MM/YYYY"
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount()),
  setEndDate: (endDate) => dispatch(setEndDate(endDate)),
  setStartDate: (startDate) => dispatch(setStartDate(startDate))
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
