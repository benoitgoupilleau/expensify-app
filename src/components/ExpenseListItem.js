import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({ description, amount, createdAt, id }) => (
  <Link className="list-item" to={`/edit/${id}`}>
    <div>
      <h3 className="list-item__title">{description}</h3>
      <span className="list-item__sub-title">{moment(createdAt).format('LL')}</span>
    </div>
    <h3 className="list-item__data">{numeral(amount / 100).format('0.00$')}</h3>
  </Link>
);

export default ExpenseListItem;

numeral.register('locale', 'fr', {
  delimiters: {
    thousands: ' ',
    decimal: ','
  },
  abbreviations: {
    thousand: 'k',
    million: 'm',
    billion: 'b',
    trillion: 't'
  },
  ordinal: (number) => (number === 1 ? 'er' : 'ème'),
  currency: {
    symbol: '€'
  }
});

numeral.locale('fr');
