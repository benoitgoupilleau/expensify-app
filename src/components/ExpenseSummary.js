import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';

import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = (props) => { // Export for test
  const expenseWord = props.expenseCount === 1 ? 'expense' : 'expenses';
  return (
    <div>
      <h1>Viewing {props.expenseCount} {expenseWord} totaling {numeral(props.expensesTotal / 100).format('0.00$')}</h1>
    </div>
  );
};

const mapStateToProps = (state) => ({
  expenseCount: selectExpenses(state.expenses, state.filters).length,
  expensesTotal: getExpensesTotal(selectExpenses(state.expenses, state.filters))
});

export default connect(mapStateToProps)(ExpenseSummary);

numeral.locale('fr');
