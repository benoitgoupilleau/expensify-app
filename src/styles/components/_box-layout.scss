.box-layout {
  align-items: center;
  background-size: cover;
  background: url('/images/bg.jpg') no-repeat center;
  display: flex;
  height: 100vh;
  justify-content: center;
  width: 100vw;
  -webkit-background-size: cover;
}

.box-layout__box {
  background: fade-out($color: #fff, $amount: 0.15);
  border-radius: 3px;
  padding: $l-size $m-size;
  text-align: center;
  width: 25rem;
}

.box-layout__title {
  line-height: 1;
  margin: 0 0 $m-size 0;
}