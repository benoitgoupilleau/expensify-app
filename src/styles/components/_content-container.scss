.content-container {
  margin: 0 auto;
  max-width: 80rem;
  padding: 0 $m-size;
}

.page-container {
  margin-bottom: $m-size;
  @media (min-width: $desktop-breakpoint) {
    margin-bottom: $l-size;
  }
}