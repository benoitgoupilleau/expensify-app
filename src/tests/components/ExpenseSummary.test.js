import React from 'react';
import { shallow } from 'enzyme';
import numeral from 'numeral';

import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';
import getExpensesTotal from '../../selectors/expenses-total';

test('should render ExpenseSummary correctly', () => {
  const expenseCount = expenses.length;
  const expensesTotal = getExpensesTotal(expenses);
  const wrapper = shallow(<ExpenseSummary expenseCount={expenseCount} expensesTotal={expensesTotal} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary correctly with 2 expenses', () => {
  const expenseCount = expenses.length;
  const expensesTotal = getExpensesTotal(expenses);
  const wrapper = shallow(<ExpenseSummary expenseCount={expenseCount} expensesTotal={expensesTotal} />);
  expect(wrapper).toMatchSnapshot();
});


numeral.register('locale', 'fr', {
  delimiters: {
    thousands: ' ',
    decimal: ','
  },
  abbreviations: {
    thousand: 'k',
    million: 'm',
    billion: 'b',
    trillion: 't'
  },
  ordinal : function (number) {
    return number === 1 ? 'er' : 'ème';
  },
  currency: {
    symbol: '€'
  }
});
