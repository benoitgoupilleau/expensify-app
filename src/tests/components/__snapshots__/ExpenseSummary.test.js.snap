// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary correctly 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totaling 
    1246,95€
  </h1>
</div>
`;

exports[`should render ExpenseSummary correctly with 2 expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totaling 
    1246,95€
  </h1>
</div>
`;
